api/test/unit/config/settings_test.js

```diff
@@ -170,6 +174,40 @@ describe('config setting Test Suite', function() {
         })
       });
     });
+
+    context('Validate cfApi', function() {
+      context('When cfApi is null', function() {
+        it('Should return false', function() {
+          settings.cfApi = null;
+          expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("cfApi is required");
+        })
+      });
+      context('When cfApi is undefined', function() {
+        it('Should return false', function() {
+          delete settings.cfApi;
+          expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("cfApi is required");
+        })
+      });
+      context('When cfApi is not a string', function() {
+        it('Should return false', function() {
+          settings.cfApi = 12345;
+          expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("cfApi must be a string");
+        })
+      });
+      it("Should add http if no protocol",function(){
+        var apiSetting = configSetting({ cfApi: defaultConfig.cfApi.replace("https://",""), db: { uri: defaultConfig.db.uri}, scheduler: { uri: defaultConfig.scheduler.uri, tls: defaultConfig.scheduler.tls }, scalingEngine: { uri: defaultConfig.scalingEngine.uri, tls: defaultConfig.scalingEngine.tls },metricsCollector: { uri: defaultConfig.metricsCollector.uri, tls: defaultConfig.metricsCollector.tls } });
```

Comment:
why not just set the cfApi to be a test url without scheme? 

The same comments for following similar test cases